testsuite/typecast/test.osl

```diff
@@ -51,4 +51,13 @@ test ()
         // point q[2] = { ... };
         // color c = (color) q[1];
     }
+
+    // Test some tricky cases -- these should not warn! The constructors
+    // ought to be able to figure out the desired type of the ambiguous
+    // function arguments.
+    closure color result = diffuse(N) * float(cellnoise(u, v));
+    color c1 = float(cellnoise(u,v));
+    color c2 = color(cellnoise(u,v));
+    color c3 = color(cellnoise(u,v),0.0,0.0);
+    color c4 = c1 * cellnoise(u,v);
```

Comment:
This last one seems ambiguous to me. `c1 * float` and `c1 * color` seem equally valid options.

Reply:
We've always disambiguated `triple OP ambig` to prefer the triple interpretation.

Reply:
So `closure color * ambig` stays `ambig` but `color * ambig` resolves to `color`?

Reply:
I should fix that, too. I will amend.

Reply:
I like the ambiguity in the `closure color` case though.

I agree that `color + ambig` should probably resolve to `color`,  but `*` and `/` seem inherently ambiguous to me, and I think it would be better to keep the warning.

Reply:
Just to be clear, this review doesn't change anything about the `ambig OP color` case.

Reply:
Sure - just brought it up since the two cases were right next to each other in the test.

As far as this particular patch, no objections at all.